workers/check_for_new_commits_on_github_worker.rb

```diff
@@ -33,6 +35,13 @@ def initialize(provider_credential: nil, project: nil)
       self.github_service = FastlaneCI::GitHubService.new(provider_credential: provider_credential)
       self.project = project
 
+      self.target_branches_set = Set.new
+      project.job_triggers.each do |trigger|
+        if trigger.kind_of?(FastlaneCI::CommitJobTrigger)
```

Comment:
We can't use the classes to distinguish between triggers? I thought that was one of the benefits of the case statement here: https://github.com/fastlane/ci/blob/062e5436416983c50740d15e0dd51f5950a30816/services/config_data_sources/json_project_data_source.rb#L31

Reply:
That's right, we don't want to use class type to distinguish between triggers. What you're seeing is actually extraction of the `.type` attribute and then the use of  `if/elsif/else` statements instead of a `case statement` in comparison with the `TRIGGER_TYPE` enumeration values against `.type`.

Check out the definition of the enumeration on https://github.com/fastlane/ci/blob/master/shared/models/job_trigger.rb#L4

JobTrigger is essentially an interface for any JobTrigger type. On that class, we have defined `.type`. So when iterating over arrays, it's best to check against the `.type` value instead of specifying the classes.

Side note, instead of using `if/elseif/else` i'd just use a `case statement` to make it clear what our intentions are:

```ruby
case trigger.type
when FastlaneCI::JobTrigger::TRIGGER_TYPE[:commit]
  job_trigger = CommitJobTrigger.from_json!(job_trigger_hash)
when FastlaneCI::JobTrigger::TRIGGER_TYPE[:nightly]
  job_trigger = NightlyJobTrigger.from_json!(job_trigger_hash)
when FastlaneCI::JobTrigger::TRIGGER_TYPE[:manual]
  job_trigger = ManualJobTrigger.from_json!(job_trigger_hash)
else
  raise "Unable to parse JobTrigger type: #{type} from #{job_trigger_hash}"
end
``

instead of 

```ruby
type = job_trigger_hash["type"]
if type == FastlaneCI::JobTrigger::TRIGGER_TYPE[:commit]
  job_trigger = CommitJobTrigger.from_json!(job_trigger_hash)
elsif type == FastlaneCI::JobTrigger::TRIGGER_TYPE[:nightly]
  job_trigger = NightlyJobTrigger.from_json!(job_trigger_hash)
elsif type == FastlaneCI::JobTrigger::TRIGGER_TYPE[:manual]
  job_trigger = ManualJobTrigger.from_json!(job_trigger_hash)
else
  raise "Unable to parse JobTrigger type: #{type} from #{job_trigger_hash}"
end
```

Reply:
note the type usage from that code, it's using:
 `if type == FastlaneCI::JobTrigger::TRIGGER_TYPE[:commit]` and not
`trigger.kind_of?(FastlaneCI::CommitJobTrigger)`

Reply:
done.